app/main/views/frameworks.py

```diff
@@ -136,6 +137,33 @@ def label_columns(labels, query):
     })
 
 
+@main.route('/frameworks/<string:framework_slug>/suppliers', methods=['GET'])
+def get_framework_suppliers(framework_slug):
+    framework = Framework.query.filter(
+        Framework.slug == framework_slug
+    ).first_or_404()
+
+    agreement_returned = request.args.get('agreement_returned')
+
+    supplier_frameworks = SupplierFramework.query.filter(
+        SupplierFramework.framework_id == framework.id
+    )
+
+    if agreement_returned is not None:
+        if convert_to_boolean(agreement_returned):
+            supplier_frameworks = supplier_frameworks.filter(
+                SupplierFramework.agreement_returned == true()
+            ).order_by(SupplierFramework.agreement_returned_at.desc())
+        else:
+            supplier_frameworks = supplier_frameworks.filter(
+                SupplierFramework.agreement_returned == false()
+            )
+
+    return jsonify(supplierFrameworks=[
+        supplier_framework.serialize() for supplier_framework in supplier_frameworks
+    ])
+
+
 @main.route('/frameworks/<string:framework_slug>/interest', methods=['GET'])
```

Comment:
Should we deprecate this `interest` endpoint in favour of the new one? (I'm not saying make all the changes now! But to tidy up at some later date.)